solr/core/src/java/org/apache/solr/handler/admin/CoreAdminHandler.java

```diff
@@ -399,19 +424,16 @@ public void shutdown() {
     if (parallelExecutor != null) ExecutorUtil.shutdownAndAwaitTermination(parallelExecutor);
   }
 
-  private static final Map<String, CoreAdminOperation> opMap = new HashMap<>();
+  private static final Map<String, CoreAdminOp> opMap = new HashMap<>();
 
-  static class CallInfo {
-    final CoreAdminHandler handler;
-    final SolrQueryRequest req;
-    final SolrQueryResponse rsp;
-    final CoreAdminOperation op;
+  public static class CallInfo {
+    public final CoreAdminHandler handler;
+    public final SolrQueryRequest req;
+    public final SolrQueryResponse rsp;
+    public final CoreAdminOp op;
 
     CallInfo(
-        CoreAdminHandler handler,
-        SolrQueryRequest req,
-        SolrQueryResponse rsp,
-        CoreAdminOperation op) {
+        CoreAdminHandler handler, SolrQueryRequest req, SolrQueryResponse rsp, CoreAdminOp op) {
```

Comment:
observations:
* making the class and its elements public is backwards compatible I'd say
* `public enum CoreAdminOperation implements CoreAdminOp {` is the case already and so the signature change from `CoreAdminOperation` to `CoreAdminOp` is also backwards compatible

Reply:
> Left couple of minor comments inline.

Thank you, it is great to hear other people opinion! 

> And next steps then could be documentation updates and/or test coverage e.g. configuring-solr-xml.adoc and solr-50-all.xml maybe?

Yes, that preferably should be done., I doesn't create tests for Solr before, for some reason when I try to run tests task it takes a huge amount of time and I have never succeeded to wait until it is ends %) If you have some nice example of tests to base it on please advice. I will try to create some if we decided to proceed with the MR.

> making the class and its elements public is backwards compatible I'd say

Yes, I was trying to keep original code as much compatible as possible not to apply too much changes...

Reply:
> ... for some reason when I try to run tests task it takes a huge amount of time and I have never succeeded to wait until it is ends ...

Yes, running the full test suite can take quite a bit of time and even if it completes not all the tests pass all the time.

> ... If you have some nice example of tests to base it on please advice ...

* Tests for the configuration being read correctly: `git grep adminHandler` finds `solr/core/src/test-files/solr/solr-50-all.xml` as a match and `git grep solr-50-all.xml` finds `solr/core/src/test/org/apache/solr/core/TestSolrXml.java` i.e. `./gradlew test --tests TestSolrXml`
* Tests for existing custom core admin handler: `git grep -l handleCustomAction` finds no match i.e. no existing test coverage it seems
* Tests for existing core admin handler: `CoreAdminHandlerTest.java` looks like a potential match and https://github.com/apache/solr/blob/releases/solr/9.0.0/solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java#L858 looks like it would provide a way to build a test without a `solr.xml` file specific to that test.